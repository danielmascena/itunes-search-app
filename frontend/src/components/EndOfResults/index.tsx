import * as React from "react";

import Box from "@mui/material/Box";
import IconButton from "@mui/material/IconButton";
import ArrowUpwardIcon from "@mui/icons-material/ArrowUpward";

const EndOfResults: React.FC = () => {
  const clickHandler = () => window.scroll(0, 0);
  return (
    <Box component="p" sx={{ textAlign: "center" }}>
      <Box component="span">End of the results. Jump to the top</Box>{" "}
      <IconButton
        color="primary"
        aria-label="Jump to the top"
        onClick={clickHandler}
      >
        <ArrowUpwardIcon />
      </IconButton>
    </Box>
  );
};

export default React.memo(EndOfResults);
