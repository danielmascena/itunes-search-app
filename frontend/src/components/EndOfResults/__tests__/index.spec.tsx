import * as React from "react";
import renderer from "react-test-renderer";

import EndOfResults from "../index";

test("renders the EndOfResults component", () => {
  expect(renderer.create(<EndOfResults />).toJSON()).toMatchSnapshot();
});
