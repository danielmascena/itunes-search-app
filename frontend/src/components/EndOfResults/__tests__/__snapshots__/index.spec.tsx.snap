// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the EndOfResults component 1`] = `
<p
  className="MuiBox-root css-xi606m"
>
  <span
    className="MuiBox-root css-0"
  >
    End of the results. Jump to the top
  </span>
   
  <button
    aria-label="Jump to the top"
    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1kuq5xv-MuiButtonBase-root-MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="ArrowUpwardIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m4 12 1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
      />
    </svg>
  </button>
</p>
`;
